piece-retriever/bin/piece-retriever.js

```diff
@@ -117,16 +131,16 @@ export default {
             egressBytes: 0,
             requestCountryCode,
             timestamp: requestTimestamp,
-            dataSetId,
+            dataSetId: retrievalCandidate.dataSetId,
             botName,
           }),
         )
         const response = new Response(
-          `Service provider ${serviceProviderId} is unavailable${retrievalResult ? ` at ${retrievalResult.url}` : ''}`,
+          `Service provider ${retrievalCandidate.serviceProviderId} is unavailable${retrievalResult ? ` at ${retrievalResult.url}` : ''}`,
           {
             status: 502,
             headers: new Headers({
-              'X-Data-Set-ID': dataSetId,
+              'X-Data-Set-ID': retrievalCandidate.dataSetId,
```

Comment:
Hmm, our X-Data-Set-ID model cannot accommodate the fact that we may have tried multiple data sets. Should we change the header to allow a list of IDs?

How about keeping the same header name `X-Data-Set-ID`, and returning a comma-separated list of IDs in the value (e.g. `10,24,40`)?


```suggestion
              'X-Data-Set-ID': allRetrievalCandidates(c => c.dataSetId).join(',')
```

I don't mind deferring this to a follow-up tech-debt issue if you prefer.

Reply:
Great catch! 9332577